src/components/cdPipeline/CDPipeline.tsx

```diff
@@ -330,8 +330,8 @@ export default class CDPipeline extends Component<CDPipelineProps, CDPipelineSta
                     ? pipelineConfigFromRes.postStageConfigMapSecretNames.secrets
                     : [],
             },
-            runPreStageInEnv: pipelineConfigFromRes.runPreStageInEnv || false,
-            runPostStageInEnv: pipelineConfigFromRes.runPostStageInEnv || false,
+            runPreStageInEnv: this.state.pipelineConfig.isVirtualEnvironment ? true : (pipelineConfigFromRes.runPreStageInEnv || false),
```

Comment:
can we create something common as it is used multiple times

Reply:
Avoiding because of last-minute change